api/v1beta1/common.go

```diff
@@ -42,3 +45,68 @@ type RolloutRestartTarget struct {
 	Kind string `json:"kind"`
 	Name string `json:"name"`
 }
+
+type Transformation struct {
+	// Templates maps a template name to its Template. Templates are always included
+	// in the rendered K8s Secret, and take precedence over templates defined in a
+	// SecretTransformation.
+	Templates map[string]Template `json:"templates,omitempty"`
+	// TransformationRefs contain references to template configuration from
+	// SecretTransformation
+	TransformationRefs []TransformationRef `json:"transformationRefs,omitempty"`
+	// Includes contains regex patterns of keys that should be included in the K8s
+	// Secret Data.
+	// FieldFilter can be used to filter the secret data that is stored in the K8s
+	// Secret Destination. Filters will not be applied to templated fields, those
+	// will always be included in the Destination K8s Secret. Exclusion filters are
+	// always applied first.
+	Includes []string `json:"includes,omitempty"`
+	// Excludes contains regex pattern for keys that should be excluded from the K8s
+	// Secret Data.
+	// FieldFilter can be used to filter the secret data that is stored in the K8s
+	// Secret Destination. Filters will not be applied to templated fields, those
+	// will always be included in the Destination K8s Secret. Exclusion filters are
+	// always applied first.
+	Excludes []string `json:"excludes,omitempty"`
+}
+
+// TransformationRef contains the configuration for accessing templates from an
+// SecretTransformation resource. TransformationRefs can be shared across all
+// syncable secret custom resources.
+type TransformationRef struct {
+	// Namespace of the SecretTransformation resource.
+	Namespace string `json:"namespace,omitempty"`
+	// Name of the SecretTransformation resource.
+	Name string `json:"name"`
+	// TemplateRefs map to a Template found in this TransformationRef. If empty, then
+	// all templates from the SecretTransformation will be rendered to the K8s Secret.
+	TemplateRefs []TemplateRef `json:"templateRefs,omitempty"`
+	// IgnoreIncludes controls whether to use the SecretTransformation's Includes
+	// data key filters.
+	IgnoreIncludes bool `json:"ignoreIncludes,omitempty"`
+	// IgnoreExcludes controls whether to use the SecretTransformation's Excludes
+	// data key filters.
+	IgnoreExcludes bool `json:"ignoreExcludes,omitempty"`
+}
+
+// TemplateRef points to templating text that is stored in a
+// SecretTransformation custom resource.
+type TemplateRef struct {
+	// Name of the Template in SecretTransformationSpec.Templates.
+	// the rendered secret data.
+	Name string `json:"name"`
+	// KeyOverride to the rendered template in the Destination secret. If Key is
+	// empty, then the Key from reference spec will be used. Set this to override the
+	// Key set from the reference spec.
+	KeyOverride string `json:"keyOverride,omitempty"`
+}
+
+// Template provides templating configuration.
+type Template struct {
+	// Name of the Template
+	Name string `json:"name,omitempty"`
```

Comment:
Related to my question in the demo's SecretTransformation, I'm wondering if `template` could be squashed down to a single field instead of always needing `template.text`.

Especially since `Transformation.Templates` and `SecretTransformationSpec.Templates` are already maps of `name:Template`.

Reply:
Great feedback, I can see that as being a possible change since the `Template.Name` is not required. Although I think the current scheme is a bit more explicit for the user, it also allows for future expansion of the `Template` definition. It also makes the template configuration consistent with `SourceTemplatesSpec.SourceTemplates`.